app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PostsController < ApplicationController
+
+  def index
+    # friends = current_user.friends.pluck(:friend_id)
+    # friends << current_user.id
+    @comment = Comment.new
+    @posts = Post.all.user_friends(current_user).with_comments_and_likes
+  end
+
+  def show
+    @post = Post.with_comments_and_likes
```

Comment:
Ok something to think about:

I log on to my Facemash account, but from the current implementation, I see all the posts of all the users of Facemash. But all I want to see are my own posts and the posts of my friends. Likewise if I visit another user's page/profile I wanna see only that user's posts and posts posted on his/her page by other users.